### ISSUE TITLE
.NET MAUI App Icon and Splash screen cannot be customized

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/maui#10079]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/NET-MAUI-App-Icon-and-Splash-screen-can/10132380)._

---
Windows 10
VS 17.3.2
iPhone 13 and Android emulator

The icon and splash screen does not change.

Steps

1: Create new .NET MAUI Project
2: Delete appicon.svg and appiconfg.svg
3: Add new appicon.svg and appiconfg.svg
4: Change appicon.svg build to MauiIcon
5: Edit the Project file with the following

          	
		

Here I keep the splash screen svg, but changed the color to show through the transparency

		
		

6: Clean Project
7: Delete the Debug and Obj folders (not that I should have to! but just to make sure there is nothing hanging around)
8: Rebuild project
9: Deploy To local IOS device.

Still shows the purple icon with .net in the middle and still shows a purple splash screen
		

---
#### Feedback Bot on 8/28/2022, 08:00 PM: 

(private comment, text removed)

---
### Original Solutions
(no solutions)

**Comment by enisn** [comment]
Did you change also `splash.svg` under `Resources/Splash` folder?

Also it's build action must be **MauiSplashScreen**

```xml
    <!-- Splash Screen -->
    <MauiSplashScreen Include="Resources\Splash\splash.svg" Color="#52D681" BaseSize="128,128" />
```

**Comment by bradyguyc** [comment]
I am not the author but have this exact issue.  I started with a blank .net maui project.   Copied a new svg file into the Resources/AppIcon and Resources/Splash director using the existing names and overwriting the file.   

When I deploy and run on a physical device attached via USB running iOS 16 (it also happened with 15).  The icon and the splash screen continued to show the default .NET icon and .NET splash screen.

When I navigate to the following directory I can see that my images has been generated in all the various sizes needed.
C:\d\TestAppIcon\TestAppIcon\obj\Debug\net6.0-ios\ios-arm64\resizetizer\r\Assets.xcassets\appicon.appiconset

![image](https://user-images.githubusercontent.com/81129882/191142068-59393f12-3679-4c01-8e50-2141fa1ca40e.png)

The only thing I changed after creating the app was coping in / over new images on top of the existing files.

I deleted the obj and bin folders and rebuilt. and then ran the app.   

No change to icon or splash screen.

Brady

**Comment by bradyguyc** [comment]
![image](https://user-images.githubusercontent.com/81129882/191142507-69ea6c65-bae7-4045-9db6-b939f1041d41.png)

![image](https://user-images.githubusercontent.com/81129882/191142546-0b123be0-3068-4423-8bf7-a6c7658ef346.png)

This is what is showing up on my physical iOS device after making the changes to the app.

**Comment by SirMangler** [comment]
I would like to add this is also an issue I've been having.

I have also tested this on Xamarin.Forms, and noted this exact same problem, despite it directly using an xasset cache instead for the iOS icons. 

Could this problem be inherent to the latest Visual Studio versions somehow?

**Comment by azig76** [comment]
I have de same problem with MAUI and ANDROID 12.

MauiSplashScreen and MauiIcon not work.

I have 5 horus and I couldn't make it work.

so I dont try more.

I hope to work with it in versions furure.

**Comment by azig76** [comment]
By the way, I'm working with Version 17.3.5.

The gif files are not animated.

**Comment by SirMangler** [comment]
Since nobody has mentioned this in the 17 days this has been open, do bare in mind you can only change the icon/splash for iOS if you're connected to a Mac (or running on a Mac), which was my issue and likely a few others here:
https://learn.microsoft.com/en-us/xamarin/xamarin-forms/deploy-test/hot-restart#limitations

**Comment by KSemenenko** [comment]
https://github.com/dotnet/maui/issues/9405
For me this is big blocker

**Comment by technik5** [comment]
> Since nobody has mentioned this in the 17 days this has been open, do bare in mind you can only change the icon/splash for iOS if you're connected to a Mac (or running on a Mac), which was my issue and likely a few others here: https://learn.microsoft.com/en-us/xamarin/xamarin-forms/deploy-test/hot-restart#limitations

I'm connected to a physical mac, but the icons/splash won't show up. Even if all png. got correctly generated as mentioned by [bradyguyc].

Could it be an error by the latest XCode-Version?

**Comment by mattleibow** [comment]
Duplicate of https://github.com/dotnet/maui/issues/8685

**Comment by 0nly1ken0bi** [comment]
Okay sooooo.. I ran into this same issue however, **I've managed to get it working** with my own custom app icon and well as my own custom splash screen.  Albeit it's not an animated splash screen, I was getting the exact same issues described above.

What I did was this.

1. Opened the Emulator and deleted the actual app from the device ( which I don't think was necessary)
2. Renamed the old SVG files to splash_old.svg and appiconfg_old.svg respectively ( which I don't think was necessary)
3. Added my custom splash.svg and appiconfg.svg files into their respective folders
4. Edited the csproj file namely removing this `Color="#512BD4" ` from this line `<MauiIcon Include="Resources\AppIcon\appicon.svg" ForegroundFile="Resources\AppIcon\appiconfg.svg"/>`
5. Run a Rebuild Solution
6. Run Build to Pixel 5 - API 33 (Android 13.0 - API 33)

Custom Icon:

![Custom_Icon](https://user-images.githubusercontent.com/54916539/198154875-60a22170-1eb5-499d-a7fc-806c3686f1f8.PNG)

Custom Splash Screen:

![Custom Splash](https://user-images.githubusercontent.com/54916539/198154953-28ce9d0c-bb99-46ed-8f04-68efbea25dd4.PNG)

Hope this helps.
